### ISSUE TITLE
Exception is rendered when NotFound is more appropriate.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3469]
|<img alt="kent_turbo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| kent_turbo reported|
|-|-|
|Trac ID|trac#3469|
|Type|enhancement|
|Created|2008-10-14 09:20:58Z|

When static.File.getChild() resolves url into a path, filepath.InsecurePath exception can occur (for example, it can occur when traversing above the resource's root, like in '/static/..'). Twisted renders 500 error with a traceback, but a simple 404 error would be more appropriate.

Attachments:

* [twisted-static-insecure.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/add/add78f1598059cb2382e033ee6270ce3d88f5b6b/69e257f7a87d26668c6e6cfbc1de7104bb0df168.patch) (566 bytes) - added by kent_turbo on 2008-10-14 09:21:07Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3469 3469
type__enhancement enhancement
reporter__kent_turbo kent_turbo
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1223976058000000 1223976058000000
changetime__1227112167000000 1227112167000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set status to `closed`|
|-|-|

(In [25445]) Merge Twisted web-static-tests-3549

Author: exarkun
Reviewer: therve
Fixes: [#3549](https://github.com/twisted/twisted/issues/3549)
Fixes: [#3469](https://github.com/twisted/twisted/issues/3469)

Add a check to `twisted.web.static.File` for security violations and turn them into 404
responses instead of letting the exception go unhandled and turn into a 500 response.

Also grab the tests for `File` from `test_web` and put them into `test_static` in a
slightly improved form.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
